x-pack/plugins/lists/server/routes/list_index/delete_list_index_route.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { validate } from '@kbn/securitysolution-io-ts-utils';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { LIST_INDEX } from '@kbn/securitysolution-list-constants';
+
+import { ListClient } from '../../services/lists/list_client';
+import type { ListsPluginRouter } from '../../types';
+import { deleteListIndexResponse } from '../../../common/api';
+import { buildSiemResponse, removeLegacyTemplatesIfExist } from '../utils';
+import { getListClient } from '..';
+
+/**
+ * Deletes all of the indexes, template, ilm policies, and aliases. You can check
+ * this by looking at each of these settings from ES after a deletion:
+ *
+ * GET /_template/.lists-default
+ * GET /.lists-default-000001/
+ * GET /_ilm/policy/.lists-default
+ * GET /_alias/.lists-default
+ *
+ * GET /_template/.items-default
+ * GET /.items-default-000001/
+ * GET /_ilm/policy/.items-default
+ * GET /_alias/.items-default
+ *
+ * And ensuring they're all gone
+ */
+export const deleteListIndexRoute = (router: ListsPluginRouter): void => {
+  router.delete(
+    {
+      options: {
+        tags: ['access:lists-all'],
+      },
+      path: LIST_INDEX,
+      validate: false,
+    },
+    async (context, _, response) => {
+      const siemResponse = buildSiemResponse(response);
+      try {
+        const lists = await getListClient(context);
+        const listIndexExists = await lists.getListIndexExists();
+        const listItemIndexExists = await lists.getListItemIndexExists();
+
+        const listDataStreamExists = await lists.getListDataStreamExists();
+        const listItemDataStreamExists = await lists.getListItemDataStreamExists();
+
+        // data streams exists, it means indices were migrated
+        if (listDataStreamExists) {
```

Comment:
I think, this could only happen, if you use patch/update listItem API, before loading Security Solution page(which performs migration to DS). Very unlikely, because, you have to use direct API call, since these APIs are not used in UI.
And those migration in patch/update was added recently, after delete method was migrated.

Thanks for noticing, I have updated the code, so it will account for this